pkg/labels/selector.go

```diff
@@ -719,10 +719,11 @@ func validateLabelValue(v string) error {
 }
 
 // SelectorFromSet returns a Selector which will match exactly the given Set. A
-// nil and empty Sets are considered equivalent to Everything().
+// nil Set is equivalent to Nothing() and an empty Sets are considered equivalent
+// to Everything().
 func SelectorFromSet(ls Set) Selector {
 	if ls == nil {
-		return internalSelector{}
+		return Nothing()
```

Comment:
Do we use nil to mean nothing anywhere?

In services, we use nil as a signal that the user is populating endpoints manually, but that's not the same thing.